File: pal.c
#include <stdio.h>

//Output true if palindrome, else false.
void main(){
    char str[500], out[500];
    printf("Enter a word/phrase to see if there is a palindrome: ");
    gets(str);
    int i = 0, j = 0;
    for(i=0;str[i] != '\0';i++){
        if(str[i] != ' '){
            out[j] = str[i];
            j++;
        }
    }
    out[j] = 0;
    int length = j-1;
    for(i=0;i <= length/2+1;i++){
        if(out[i] != out[length-i]){
            printf("False\n");
            return;
        }
    }
    printf("True\n");
}

File: pal.py
import re, string

regex = re.compile('[^a-zA-Z]')
x = raw_input("Enter phrase to check for palindrome: ")

#figure out if the current string is a palindrome using a naive reversal.
#alternatively, punctuation could be handled by stripping the string
#of all characters that are not a-zA-Z1-9

retval = regex.sub('', x)
if retval == retval[::-1]:
    print retval, " is a palindrome.\n" 
else:
    print("Not a palindrome.\n")
